Automattic/wp-calypso
# Expose Hosting Trial: Allow starting free trial from themes upload page

Related to https://github.com/Automattic/dotcom-forge/issues/5097

## Proposed Changes

This PR allow the user to start a free trial from the plugins upload page.  There is an issue where we gets redirected to crowdsignal after the atomic transfer. what should happen is that we should be redirected to `/wp-admin/theme-install.php`

* Show free trial banner when user is eligible
* Shows the trial acknowledge modal and start the trial

## Testing Instructions



* go to `/themes/upload/simple_site`
* Start your free trial and verify the site transitions to atomic successfully 

![image](https://github.com/Automattic/wp-calypso/assets/47489215/75107a2d-a2c3-4b7a-8d86-8408f7ecc8a9)


## Pre-merge Checklist
